scikits/odes/sundials/cvode.pyx

```diff
@@ -1180,22 +1205,66 @@ cdef class CVODE:
         np_tspan = np.asarray(tspan, dtype=float)
         np_y0    = np.asarray(y0, dtype=float)
 
-        return self._solve(np_tspan, np_y0)
+        soln = self._solve(np_tspan, np_y0)
+
+        flag_no, t, y, t_err, y_err = soln[:5]
+
+        if soln[5]:
+            t_roots = np.array(soln[5])
+        else:
+            t_roots = None
+
+        if soln[6]:
+            y_roots = np.array(soln[6])
+        else:
+            y_roots = None
+
+        if soln[7]:
+            t_tstop = np.array(soln[7])
+        else:
+            t_tstop = None
+
+        if soln[8]:
+            y_tstop = np.array(soln[8])
+        else:
+            y_tstop = None
+
+        flag = self.RETURN_FLAGS.get(flag_no, "UNKNOWN")
+
+        soln = CVODE_Return(
+            flag=flag, flag_no=flag_no, values=(t,y), err_values=(t_err,y_err),
+            root_values=(t_roots,y_roots), tstop_values=(t_tstop,y_tstop)
+        )
+
```

Comment:
All the conversion above is needless if the user wants to use the old API.
Also maybe a one-liner would be in this case more readable e.g.
```python
t_stop = np.array(soln[7]) if soln[7] else None
```
Would it not be simpler to do all of this directly in the _solve() method and just return whatever _solve() returns - as it was before? Or does it add too much overhead to the cython function so that it's not feasible?